extensions/ql-vscode/src/view/raw-results-table.tsx

```diff
@@ -27,14 +27,19 @@ export class RawTable extends React.Component<RawTableProps, {}> {
       dataRows = dataRows.slice(0, RAW_RESULTS_LIMIT);
     }
 
-    const tableRows = dataRows.map((row: ResultRow, rowIndex: number) =>
-      <RawTableRow
-        key={rowIndex}
-        rowIndex={rowIndex + this.props.offset}
-        row={row}
-        databaseUri={databaseUri}
-      />
-    );
+    const emptyTableRow = <tr key={0}><td key={-1}>
```

Comment:
Can you extract this into a function and only call it down below so that we are not generating this virtual DOM unless it's needed?

Reply:
Also, this is rendering the message as a row in the table. If there are no results, we shouldn't even display the table. Take a look at `renderNoResults` in `alert-table.tsx`.  This basically does what I'm suggesting, though the message here will also need to be updated similarly.

Reply:
Thanks for the initial review! Just to clarify, should I also change the message returned in `renderNoResults()` in `alert-table.tsx`? For example, replace `No Alerts` in `return <span>No Alerts</span>;` to a similar message as I'm adding in `raw-results-table.tsx`.

Reply:
Yes, please.

Reply:
Hi @aeisenberg, as suggested I've extracted the message into a function that is called in `alert-table.tsx` and `raw-results-table.tsx` if there are are no query results.